scripts/ms

```diff
@@ -82,28 +63,79 @@ def init(opts):
-                 "nprocs": opts.n_processes,
                  "method": method,
                  "out_dir": out_dir,
                  "out_state": out_state,
                  "state": state_file,
                  "domain": domain_file,
                  "forcing": forcing_files,
-                 "verbose": opts.verbose * logging.INFO})
+                 "verbose": verbose * logging.INFO})
     conf['out_vars'] = to_list(conf.get('out_vars', '[]'))
     conf['iter_dims'] = to_list(conf.get('iter_dims', '["lat", "lon"]'))
-    if opts.time_grouper is not None:
-        conf['time_grouper'] = opts.time_grouper
+    conf['time_grouper'] = time_grouper if time_grouper else None
     conf = {k: v for k, v in conf.items() if v != []}
     return conf
 
 
-def main():
+def setup_dask(scheduler, num_workers):
+
+    if scheduler is None and num_workers is None:
+        return None
+
+    if scheduler is None and num_workers:
+        scheduler = 'multiprocessing'  # default when num_workers is not None
+
+    import dask
+    options = {}
+    if os.path.isfile(scheduler):
+        from dask.distributed import Client
+        client = Client(scheduler_file=scheduler)
+        options['get'] = client.get
+        scheduler = 'distributed'
+    elif scheduler == 'distributed':
+        from dask.distributed import Client
+        client = Client(n_workers=num_workers, diagnostics_port=8787)
+        logger.info('Dask Client: \n %s \n ' % client)
+        options['get'] = client.get
+    elif scheduler == 'multiprocessing':
+        from multiprocessing import Pool
+        import dask.multiprocessing
+        options['get'] = dask.multiprocessing.get
+        if num_workers is not None:
+            options['pool'] = Pool(num_workers)
+    elif scheduler == 'threaded':
+        from multiprocessing.pool import ThreadPool
+        import dask.threaded
+        options['get'] = dask.threaded.get
+        if num_workers is not None:
+            options['pool'] = ThreadPool(num_workers)
+    elif scheduler == 'synchronous':
+        options['get'] = dask.get
+    else:
+        raise ValueError('unknown scheduler %s' % scheduler)
+
+    return dask.set_options(**options)
+
+
+@click.command(context_settings=dict(help_option_names=['-h', '--help']))
+@click.argument('config', type=click.Path(exists=True))
+@click.option('--scheduler', type=str, default=None,
+              help='Parallel dask scheduler to use, defaults to '
+                   'multiprocessing when num_workers is set.')
+@click.option('-n', '--num_workers', default=None, type=int,
+              help='Number of workers to use during parallel computations')
+@click.option('-tg', '--time_grouper', default=None, type=str, multiple=True,
+              help='Pandas TimeGrouper string (e.g. `1AS`)')
+@click.option('-v', '--verbose', count=True,
+              help='Set the Verbosity of MetSim')
+def main(config, scheduler, num_workers, time_grouper, verbose):
```

Comment:
@arbennett - any comments on this user interface?

Reply:
I think it might be useful to list out the main schedulers available for use in the help, but other than that I like it.

Reply:
done in 08f8199